File: problem/0344. Reverse String.js
// 344. Reverse String
// https://leetcode.com/problems/reverse-string/
// Tag: Array, String, Easy

/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */

var reverseString = function(s) {
    for(let i=0; i<s.length/2; i++){
        [s[i], s[s.length-1-i]] = [s[s.length-1-i], s[i]];
    }
};

// --- for testing ---
let arr = ["h","e","l","l","o"]; // Output: ["o","l","l","e","h"]
reverseString(arr);
console.log(arr); // ["o","l","l","e","h"]

File: c#-solution/0344. Reverse String.cs
// 0344 Reverse String
// https://leetcode.com/problems/reverse-string/description/
// Tag: Array, Two Pointers, String, Easy

public class Solution {
    public void ReverseString(char[] s) {
        int haf = s.Length/2;

        for(int i=0; i<haf; i++){
            var c = s[i];
            s[i] = s[s.Length -1 - i];
            s[s.Length -1 -i] = c;
        }
    }
}

// use span
public class Solution {
    public void ReverseString(char[] s) {
        s.AsSpan().Reverse();
    }
}